### PR TITLE
ceres-solver: adding version 2.2.0

### PR DESCRIPTION
**Pull request by snehring**
Version bump

### CODE REVIEW

File: var/spack/repos/builtin/packages/ceres-solver/package.py [hunk 1]
```diff
@@ -25,12 +26,25 @@ class CeresSolver(CMakePackage):
...
     depends_on("eigen@3:")
+    depends_on("eigen@3.3:", when="@2.0.0:")
     depends_on("lapack")
-    depends_on("glog")
+    depends_on("glog@0.3.5:")
 
     def cmake_args(self):
-        args = ["-DCXSPARSE=OFF", "-DEIGENSPARSE=ON", "-DLAPACK=ON", "-DSCHUR_SPECIALIZATIONS=OFF"]
+        args = []
+        if self.spec.satisfies("@:2.0.0"):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/ceres-solver/package.py:40]: Curious. 

Why are the rest of these options no longer relevant starting with v2.2.0 (aside from perhaps `CXSPARSE`) or are you relying on defaults?

Are the docs outdated at http://ceres-solver.org/installation.html#options-controlling-ceres-configuration?

**Review by tldahlgren — COMMENTED** [review]
Confirmed the version sha256.  The rest LGTM though some questions about `cmake_args`.

File: var/spack/repos/builtin/packages/ceres-solver/package.py (hunk 1 above)
Comment (snehring) [reply var/spack/repos/builtin/packages/ceres-solver/package.py:40]: CXSPARSE is no longer present in the cmake config, everything else defaults to ON. I don't understand why SCHUR_SPECIALIZATIONS was being disabled, but in the interest of not changing the behavior of 2.0.0 and below I decided to preserve build behavior for the older versions. What prompted the changes originally was a failure to build 2.2.0 at least +suitesparse (which comes highly recommended, enough that I was tempted to make the variant default=True).

**Comment by snehring** [comment]
Actually, this seems to not actually be detecting suitesparse when +suitesparse, so it still needs some work.

**Comment by snehring** [comment]
probably the lack of any suite-sparse dependency!

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**